### PR TITLE
Fetch config files from GitHub

### PR DESCRIPTION
**Pull request by PLASMAchicken**
This PR adds a Function that onServer start checks for the files needed by Essentials and if needed downloads them from GitHub

### CODE REVIEW

File: src/BPEssentials/FileChecker.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public static async Task CheckFiles()
             }
             foreach (var file in RequiredFilesDictionary)
             {
-                if (File.Exists(file.Value)) continue;
+                if (File.Exists(file.Value)) { continue; }
```
Comment (UserR00T) [inline src/BPEssentials/FileChecker.cs]: Please put `{` and `}` on new lines. It doesn't look as good, just makes it clearer to read.

```suggestion
                if (File.Exists(file.Value))
                {
                    continue;
                }
```

File: src/BPEssentials/Core.cs [hunk 2]
```diff
@@ -119,9 +119,10 @@ public void ReadConfigurationFiles()
             CustomCommandsReader.ReadAndParse();
         }
 
-        public void OnReloadRequest()
+        public async void OnReloadRequestAsync()
```
Comment (UserR00T) [inline src/BPEssentials/Core.cs]: An `async` method with `void` as return is not recommended. Consider changing it to `Task`. Might not be possible in current scenario, let me know.

```suggestion
        public async Task OnReloadRequestAsync()
```

File: src/BPEssentials/FileChecker.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace BPEssentials
+{
+    public static class FileChecker
+    {
+        public static readonly HttpClient client = new HttpClient();
```
Comment (UserR00T) [inline src/BPEssentials/FileChecker.cs]: Suggested to only use properties and private fields. Consider changing to a property (getter only).

```suggestion
        public static HttpClient Client { get; }= new HttpClient();
```

File: src/BPEssentials/FileChecker.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                Core.Instance.Logger.LogWarning("Essentials Folder not found.");
+                Directory.CreateDirectory(Paths.EssentialsFolder);
+                Core.Instance.Logger.LogInfo("Created Essentials Folder.");
+            }
+            foreach (var file in RequiredFilesDictionary)
+            {
+                if (File.Exists(file.Value)) continue;
+                try
+                {
+                    Core.Instance.Logger.LogWarning($"{file.Key} not found.");
```
Comment (UserR00T) [inline src/BPEssentials/FileChecker.cs]: Inform the user that the file is going to be downloaded.
```suggestion
                    Core.Instance.Logger.LogWarning($"{file.Key} was not found; downloading.");
```

File: src/BPEssentials/FileChecker.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                    var content = await client.GetStringAsync($"{Core.Instance.Info.Github}/raw/{(Core.IsDevelopmentBuild() ? "development" : "master")}/dist/{file.Key}");
+                    File.WriteAllText(file.Value, content);
+                    Core.Instance.Logger.Log($"{file.Key} was downloaded.");
+                    client.Dispose();
+                }
+                catch (HttpRequestException e)
+                {
+                    Core.Instance.Logger.LogError($"{file.Key} could not be donwloaded because of: {e.Message}.");
+
```
Comment (UserR00T) [inline src/BPEssentials/FileChecker.cs]: Remove extra new line.
```suggestion
```

File: src/BPEssentials/FileChecker.cs [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                {
+                    Core.Instance.Logger.LogWarning($"{file.Key} not found.");
+                    var content = await client.GetStringAsync($"{Core.Instance.Info.Github}/raw/{(Core.IsDevelopmentBuild() ? "development" : "master")}/dist/{file.Key}");
+                    File.WriteAllText(file.Value, content);
+                    Core.Instance.Logger.Log($"{file.Key} was downloaded.");
+                    client.Dispose();
+                }
+                catch (HttpRequestException e)
+                {
+                    Core.Instance.Logger.LogError($"{file.Key} could not be donwloaded because of: {e.Message}.");
```
Comment (UserR00T) [inline src/BPEssentials/FileChecker.cs]: Might want to log the stacktrace somewhere. Maybe we should utilize an `exceptions.txt` file again? Maybe as json list?

File: src/BPEssentials/FileChecker.cs [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+            foreach (var file in RequiredFilesDictionary)
+            {
+                if (File.Exists(file.Value)) continue;
+                try
+                {
+                    Core.Instance.Logger.LogWarning($"{file.Key} not found.");
+                    var content = await client.GetStringAsync($"{Core.Instance.Info.Github}/raw/{(Core.IsDevelopmentBuild() ? "development" : "master")}/dist/{file.Key}");
+                    File.WriteAllText(file.Value, content);
+                    Core.Instance.Logger.Log($"{file.Key} was downloaded.");
+                    client.Dispose();
```
Comment (UserR00T) [inline src/BPEssentials/FileChecker.cs]: `client` cannot be disposed in a `foreach` loop. consider removing this line.
```suggestion
```

File: src/BPEssentials/FileChecker.cs [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+            if (!Directory.Exists(Paths.EssentialsFolder))
+            {
+                Core.Instance.Logger.LogWarning("Essentials Folder not found.");
+                Directory.CreateDirectory(Paths.EssentialsFolder);
+                Core.Instance.Logger.LogInfo("Created Essentials Folder.");
+            }
+            foreach (var file in RequiredFilesDictionary)
+            {
+                if (File.Exists(file.Value)) continue;
```
Comment (UserR00T) [inline src/BPEssentials/FileChecker.cs]: Please put `{` and `}` on new lines. It doesn't look as good, just makes it clearer to read.

```suggestion
                if (File.Exists(file.Value))
                {
                      continue;
                }
```

File: src/BPEssentials/FileChecker.cs [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+                }
+                catch (HttpRequestException e)
+                {
+                    Core.Instance.Logger.LogError($"{file.Key} could not be donwloaded because of: {e.Message}.");
+
+                }
+            }
+        }
+
+        private static Dictionary<string, string> RequiredFilesDictionary = new Dictionary<string, string>()
```
Comment (UserR00T) [inline src/BPEssentials/FileChecker.cs]: Make this a property. Fields should not be PascalCased.
```suggestion
        private static Dictionary<string, string> RequiredFilesDictionary { get: }= new Dictionary<string, string>()
```

Comment (UserR00T) [inline src/BPEssentials/FileChecker.cs]: Consider pushing this type up [according to the C# spec.](https://stackoverflow.com/a/310967/6244263)

File: src/BPEssentials/Core.cs (hunk 2 above)
Comment (PLASMAchicken) [reply src/BPEssentials/Core.cs]: This is not possible due to `new Action()` only accepting void.

Comment (UserR00T) [reply src/BPEssentials/Core.cs]: Alright. We'll figure this out later on.

**Pull request merged**